pkg/dist/integration_types.yml

```diff
@@ -551,9 +554,7 @@ kafka_mirrormaker:
         consumer_auto_offset_reset:
           title: Set the auto.offset.reset to consumer.
           description: 'Set where consumer starts to consume data. Value `earliest`: Start replication from the earliest offset. Value `latest`: Start replication from the latest offset. Default is `earliest`.'
-          type:
-            - string
-            - "null"
```

Comment:
Is this safe? I wonder if `null` has a specific meaning for the backend

Reply:
Yeah. Golang doesn't support both optional and nullable fields, e.g.:
```go
type Foo struct {
    Nullable *string `json:"nullable"` // go won't omit nil
    Optional *string `json:"optional,omitempty"` // go omits nil
    NonNull   string  `json:"nonnull,omitempty" // non-nullable, omits ""
    Required  string  `json:required" // sends ""
}
```
So there is no way (for structs) to have "optional nullable string" (like in your example).
Technically we shouldn't have such things in the API.

Reply:
Also, please have a look how does affect tf generator https://github.com/aiven/terraform-provider-aiven/pull/1887/files